spec/integration/associations/has_one_spec.rb

```diff
@@ -173,4 +173,42 @@
       end
     end
   end
+
+  context "when assigning the same value to a belongs_to" do
+    let(:post) { HomPost.create! }
+    let(:comment) { HomComment.create! }
+    let(:comment2) { HomComment.create! }
+
+    it "persists it correctly" do
+      post.comment = comment
+      post.reload
+      expect(post.comment).to eq(comment)
+
+      post.comment = comment
```

Comment:
Can you add the following test case please:

```
comment2 = Comment.new(id: comment.id)
post.comment = comment2
```

Reply:
I added this test case and it fails with a dup key exception. This case attempts to insert a new comment (since its new record attribute is true) into the comments collection, and fails since it has the same _id. 

You're right that the `replacement != self` condition does fail, but I think that's okay in this case, since the use case of trying to add a document with the same _id doesn't actually make sense, and we shouldn't go forward with unbinding and rebinding etc. If you want to modify the comment, you should get it from post.comment and update it... 

I think all in all the functionality is correct for this case, we should raise an error and not do anything.

Reply:
After thinking this over I think you're right... The only downside to this approach are the extra operations that happen when assigning the same document, that don't share the same reference, twice.

Reply:
see my comment below